### ISSUE TITLE
floatting overflow with land test on pm-cpu/intel in NitrifDenitrifMod.F90

### ISSUE THREAD

**Issue by ndkeen** [E3SM-Project/E3SM#5767]
I see this fail on next with CDASH testing.
`ERS_D.f19_g16.I1850GSWCNPRDCTCBC.pm-cpu_intel.elm-ctc_f19_g16_I1850GSWCNPRDCTCBC`

I think it has failed this way at least one other time.

```
 57: forrtl: error (72): floating overflow
 57: Image              PC                Routine            Line        Source
 57: libpthread-2.31.s  00001520B4BA68C0  Unknown               Unknown  Unknown
 57: e3sm.exe           0000000005AFECEE  Unknown               Unknown  Unknown
 57: e3sm.exe           0000000003062031  nitrifdenitrifmod         267  NitrifDenitrifMod.F90
 57: e3sm.exe           000000000341E2BD  soillittdecompmod         381  SoilLittDecompMod.F90
 57: e3sm.exe           0000000002EE0D1D  ecosystemdynmod_m         556  EcosystemDynMod.F90
 57: e3sm.exe           00000000009839C0  elm_driver_mp_elm        1101  elm_driver.F90
 57: e3sm.exe           000000000090E914  lnd_comp_mct_mp_l         514  lnd_comp_mct.F90
 57: e3sm.exe           0000000000491D95  component_mod_mp_         751  component_mod.F90
 57: e3sm.exe           00000000004531A2  cime_comp_mod_mp_        2904  cime_comp_mod.F90
 57: e3sm.exe           000000000047A722  MAIN__                    153  cime_driver.F90
```

**Comment by peterdschwartz** [comment]
Yeah, this is a new test from @bishtgautam 's PR that added new debug tests.  The corresponding test passes on chrysalis intel, so the different compiler flags for pm must be finding something.  I'll look into it now that perlmutter is online

**Comment by peterdschwartz** [comment]
This is fixed by PR #5780

**Comment by bishtgautam** [comment]
Reopening this issue because #5780 missed one more required change that is now added in #5828
